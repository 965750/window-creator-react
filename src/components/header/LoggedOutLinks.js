import React from 'react'
import styles from './LoggedOutLinks.module.scss'
import { FormattedMessage } from 'react-intl'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'

const LoggedOutLinks = () => {
    return (
        <div className={`${styles.loggedIn__wrapper} uppercase flex`}>
            <NavLink
                exact
                to="/register"
                className="text-darkText hover:text-malachite flex self-center cursor-pointer mr-10"
                activeClassName={styles['link--active']}
            >
                <FormattedMessage id="RegisterNav" defaultMessage="Register" />
            </NavLink>
            <NavLink
                exact
                to="/"
                className="text-darkText hover:text-malachite flex self-center cursor-pointer mr-10"
                activeClassName={styles['link--active']}
            >
                <FormattedMessage id="LoginNav" defaultMessage="Login" />
            </NavLink>
        </div>
    )
}

LoggedOutLinks.propTypes = {
    // logout: PropTypes.func,
}

export default LoggedOutLinks
