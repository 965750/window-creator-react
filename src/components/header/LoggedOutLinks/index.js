import { NavLink, BrowserRouter } from 'react-router-dom'
import { FormattedMessage } from 'react-intl'

import React from 'react'
import styles from './LoggedOutLinks.module.scss'

const Index = () => (
  <div className="uppercase flex">
    <BrowserRouter>
      <NavLink
        exact
        to="/register"
        className="text-osloGray hover:text-malachite flex self-center cursor-pointer sm:mr-10 mr-4"
        activeClassName={styles['link--active']}
      >
        <FormattedMessage id="RegisterNav" defaultMessage="Register" />
      </NavLink>
      <NavLink
        exact
        to="/"
        className="text-osloGray hover:text-malachite flex self-center cursor-pointer"
        activeClassName={styles['link--active']}
      >
        <FormattedMessage id="LoginNav" defaultMessage="Login" />
      </NavLink>
    </BrowserRouter>
  </div>
)

export default Index
