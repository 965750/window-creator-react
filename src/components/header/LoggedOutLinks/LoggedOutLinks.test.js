import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux'
import { BrowserRouter } from 'react-router-dom'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react'

import LoggedOutLinks from './index'
import messages from '../../../messages-i18n'
import thunk from 'redux-thunk';
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
  firebase: {
    auth: {
      email: 'mockEmail@gmail.com'
    },
    profile: {
      firstName: 'mockName',
      lastName: 'mockLastName',
      isLoaded: true,
      isEmpty: false,
      windows: [{
        id: 'mockId',
        save: {
          name: 'mockWindowName'
        }
      }]
    }
  }
}
const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('LoggedOutLinks', () => {
  let wrapper
  const initialProps = {}

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <BrowserRouter>
        <IntlProvider
          locale="en"
          messages={messages[store.getState().theme.lang]}
        >
          <LoggedOutLinks {...props} />
        </IntlProvider>
      </BrowserRouter>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
