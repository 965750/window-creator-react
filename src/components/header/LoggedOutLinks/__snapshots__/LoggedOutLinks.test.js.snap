// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedOutLinks should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="uppercase flex"
  >
    <a
      class="text-osloGray hover:text-malachite flex self-center cursor-pointer sm:mr-10 mr-4"
      href="/register"
    >
      Rejestracja
    </a>
    <a
      aria-current="page"
      class="text-osloGray hover:text-malachite flex self-center cursor-pointer link--active"
      href="/"
    >
      Logowanie
    </a>
  </div>
</DocumentFragment>
`;
